app/lib/scorecard/backend.dart

```diff
@@ -218,23 +219,68 @@ class ScoreCardBackend {
         if (size > _reportSizeDropThreshold) {
           _logger.reportError(
               '$reportType report exceeded size threshold ($size > $_reportSizeWarnThreshold)');
-          // Size trimming is temporarily disabled.
-          // TODO: re-enable after fixing the underlying issues with the verbose reports and package:appengine.
-          return false;
+          return true;
         } else if (size > _reportSizeWarnThreshold) {
           _logger.warning(
               '$reportType report exceeded size threshold ($size > $_reportSizeWarnThreshold)');
         }
         return false;
       }
 
-      if (sizeCheck(ReportType.pana, scoreCard.panaReportJsonGz)) {
-        // TODO: replace with something meaningful
-        scoreCard.panaReportJsonGz = <int>[];
+      if (panaReport != null &&
+          sizeCheck(ReportType.pana, scoreCard.panaReportJsonGz)) {
+        scoreCard.updateReports(
+          panaReport: PanaReport(
+            timestamp: DateTime.now().toUtc(),
+            panaRuntimeInfo: panaReport.panaRuntimeInfo,
+            reportStatus: panaReport.reportStatus,
+            derivedTags: panaReport.derivedTags?.take(100).toList(),
+            allDependencies: panaReport.allDependencies?.take(100).toList(),
+            licenseFile: panaReport.licenseFile,
+            report: pana.Report(
+              sections: panaReport.report?.sections
+                      .map(
+                        (s) => s.summary.length < 1000
+                            ? s
+                            : pana.ReportSection(
+                                id: s.id,
+                                title: s.title,
+                                status: s.status,
+                                grantedPoints: s.grantedPoints,
+                                maxPoints: s.maxPoints,
+                                summary: [
+                                  s.summary.substring(0, 900),
```

Comment:
This is fine, but if we wanted to be really advanced we could take 500 chars from the beginning, and 400 from the end. The end might have some relevant information about what went wrong.

Reply:
Can we run into unicode issues here when we do the raw substring? Not sure how important it is to handle, but I think this might split utf-16 4-byte encoded characters.

Reply:
+1 on the beginning + end split. Unicode charcodes could be an issue, but IIRC it would not cause issues besides the character involved (which may be an issue we can ignore for now).